ClientRuntimes/NodeJS/ms-rest-azure/lib/azureServiceClient.js

```diff
@@ -57,73 +57,76 @@ function AzureServiceClient(credentials, options) {
 util.inherits(AzureServiceClient, msRest.ServiceClient);
 
 /**
- * Poll Azure long running PUT operation.
- * @param {object} [resultOfInitialRequest] - Response of the initial request for the long running operation.
+ * Poll Azure long running GET, PATCH, POST or DELETE operations.
+ * @param {object} [resultOfInitialRequest] - result of the initial request.
  * @param {object} [options]
  * @param {object} [options.customHeaders] headers that will be added to request
  */
-AzureServiceClient.prototype.getPutOrPatchOperationResult = function (resultOfInitialRequest, options, callback) {
+AzureServiceClient.prototype.getLongRunningOperationResult = function (resultOfInitialRequest, options, callback) {
   var self = this;
-
-  if(!callback && typeof options === 'function') {
+  
+  if (!callback && typeof options === 'function') {
     callback = options;
     options = null;
   }
   if (!callback) {
     throw new Error('Missing callback');
   }
-  
+
   if (!resultOfInitialRequest) {
     return callback(new Error('Missing resultOfInitialRequest parameter'));
   }
   
-  if (resultOfInitialRequest.response.statusCode !== 200 &&
-      resultOfInitialRequest.response.statusCode !== 201 &&
-      resultOfInitialRequest.response.statusCode !== 202) {
-    return callback(new Error(util.format('Unexpected polling status code from long running operation \'%s\'', 
-      resultOfInitialRequest.response.statusCode)));
+  if (!resultOfInitialRequest.response) {
+    return callback(new Error('Missing resultOfInitialRequest.response'));
+  }
+
+  if (this._checkInitialRequestResponseStatusCodeFailed(resultOfInitialRequest)) {
+    return callback(new Error(util.format('Unexpected polling status code from long running operation \'%s\' for method \'%s\'',
```

Comment:
@csmengwan I see a scenario that worked in the old implementation that looks like it doesn't work here.  What if the initial response to a PUT operation is a status code 200 with ProvisioningState == Failed.  In the old case, this would return an error with provisioning failed.  In the new case this will return 'unexpected status code'.  This looks like it's a result of the way the code was restructured